src/system/SystemPacketBuffer.h

```diff
@@ -730,10 +730,9 @@ class PacketBufferWriterBase : public Writer
      *  @param[in]  aAvailableSize  Length bound of the BufferWriter.
      *  @param[in]  aReservedSize   Reserved packet buffer space for protocol headers; see \c PacketBufferHandle::New().
      */
-    PacketBufferWriterBase(size_t aAvailableSize, uint16_t aReservedSize = CHIP_SYSTEM_CONFIG_HEADER_RESERVE_SIZE) :
-        Writer(nullptr, 0)
+    PacketBufferWriterBase(System::PacketBufferHandle && aPacket) : Writer(aPacket->Start(), aPacket->AvailableDataLength())
```

Comment:
Looks like this still needs a length (possibly optional; checking the cases), for the case that the pool-backed buffer is actually larger than requested.